apps/src/lib/levelbuilder/shapes.jsx

```diff
@@ -93,24 +95,24 @@ export const levelShapeForScript = PropTypes.shape({
 
 export const lessonShape = PropTypes.shape({
   id: PropTypes.number,
-  key: PropTypes.string,
+  key: PropTypes.string.isRequired,
   name: PropTypes.string,
-  position: PropTypes.number,
+  position: PropTypes.number.isRequired,
```

Comment:
I imagine there are some which are hard to make required for lessonShape, scriptLevelShape and levelShape because these are shared between script edit and lesson edit. thinking on this more, I think the direction we should be moving in is to have separate shapes for script edit and lesson edit. This will match the pattern used on the backend where there is a different summarize method for script edit and lesson edit. Happy to chat about this if you have a different vision as there are definitely alternatives.

Reply:
I definitely think that would be ideal. I'm not sure if we want to spend the time to do that work right now. What are your thoughts?

Reply:
agreed, let's not do that work right now. if we do end up needing to make any changes, that's the direction I'd want to head in. 

Once we do [PLAT-460](https://codedotorg.atlassian.net/browse/PLAT-460) we won't even need scriptLevelShape and levelShape on the script edit page, so that's a good reason not to invest time there now. when we get there, I might push harder for splitting lessonShape to be separate for script and lesson edit pages, though, since that could help make a clear division between what attributes are owned by each edit page.